web/index.html

```diff
@@ -376,32 +437,36 @@ <h3>Used in</h3>
             return queryParams;
         }
 
-        // Decode base64 value mod 1000
-        function getChunk(value) {
-            let base64 = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-='
-            let result = 0
-            for (let i = 0; i < value.length; i++) {
-                result = (result * 64 + base64.indexOf(value[i]))
-            }
-            return Math.floor(result / 100)
-        }
-
-        const DATA_PREFIX = 'data/';
+        const PATH_PREFIX = window.location.host == 'expitau.github.io' ? '/' : '';
+        const CHUNK_WORKER_PATH = `${PATH_PREFIX}loadChunkWorker.js`;
+        const DATA_PATH = `${PATH_PREFIX}data/`;
+        const HELPER_PATH = window.location.host == 'expitau.github.io' ? 'https://expitau.github.io/InfiniteCraftWiki/helper'
+            : 'http://127.0.0.1:5501/web/helper.html';
 
         // Vue app
         const app = new Vue({
             el: '#app',
             data: {
                 data: { index: {}, list: [], costs: {}, craftingCache: {}, data: {}, metadata: { recipeCount: 0, loadedChunks: [] } },
+                chunkLoadState: [],
+                craftingTreeData: null,
+                bookmarkletData: "",
                 currentElement: "",
+                elementLoaded: false,
                 showHiddenRecipes: false,
                 searchQuery: "",
                 showToast: false,
-                saveData: [],
+                showImportToast: false,
+                isDraggingBookmarklet: false,
+                isShaking: false,
+                saveData: new Set(),
+                saveDataArray: [],
```

Comment:
save data is now stored and used primarily in Set form instead of as an Array.
The array is what is received from the url or cookies, and is kept around so that it can easily be passed to workers in that array form, but by also storing it as a set, it allows for O(1) read times for the saveData.has() checks, instead of the O(n) checks for saveDataArray.includes(). Since this is used for every single element that is rendered, this increased performance a good deal.